core-wildfly/src/main/java/sunstone/annotation/DomainMode.java

```diff
@@ -15,4 +15,6 @@
     String user() default "${" + WildFlyConfig.MGMT_USERNAME + "}";
     String password() default "${" + WildFlyConfig.MGMT_PASSWORD + "}";
     String port() default "${" + WildFlyConfig.MGMT_PORT + "}";
+    //comma separated list
+    String serverList() default "${" + WildFlyConfig.DOMAIN_SERVERS + "}";
```

Comment:
that would be preferable, but I'm not aware of simple way of doing that, because it also has to accept string format from `sunstone.properties`

Reply:
Java allows  `String[] servers() default "${sunstone.wildfly.domain.servers}";`
Then, you can have `servers = {"1", "2"}` or just `servers = "1"` or leave it to the default value. Now, the question is should we allow a user to set `sunstone.wildfly.domain.servers` to a list? Perhaps, it would be the first case in the sunstone and the property name suggests it is allowed. Take a look on this https://download.eclipse.org/microprofile/microprofile-config-3.0/microprofile-config-spec-3.0.html#configexamples

Reply:
implementing serverList as array seems to introduce unnecessary complexity it would look like this
```java
deployToEc2Instance(...) {
String[] serverGroupsParams = wildFly.domain().servers();
for (serverParam : serverGroupsParams){
      String[] sgName = resolveParamToList(serverParam)
      serverGroupsNames.stream().map(Optional::get).forEach(builder::toServerGroups);
      if sgName.iSEmpty()
           builder.toAllGroups()
```
because we have to resolve each array index separately and default value ( which may be list in properties) is resolved to array of length 1 in `wildFly.domain().servers()`

Reply:
there is a elegant solution, by just changing:
` String serverGroupsParam = String.join(",", wildFly.domain().servers()))`
in my current solution

Reply:
I understand, but the priority is the usability of the annotation, not the complexity of the handler.

Using it like `servers = {"${my property}", "other-group"}` where `my.property` is another comma separated list isn't really a best practice. But if someone needs it, then so be it.

I am afraid using comma separated list as a value in annotation isn't a best practice as well and IMO isn't intuitive.  Why would a user expect such API when annotations provide an easy option to define a list of strings?

Moreover what would be the best name for parameter? `serverGroups`? I think `server` could lead to a idea that it deploys to a specific server not a server group.

Note: in your example, calling `builder.toAllGroups()` will be called in this scenario `servers = {"", "some-group"}`. Which is really weird and should at least print a warning.

Reply:
i do agree, list is better i chose second implementation. serverGroups sounds good. I was aware of your note, but didn't want to make code snippet too complex